packages/protocol/contracts/psp/README.md

```diff
@@ -0,0 +1,23 @@
+## PSP Notes
+
+Currently, most DeFi products depend on a live data point provided by a passive oracle service, and keepers to trigger actions automatically based on the data provided by the oracle, e.g., liquidate a user that has shorted an asset that has increased in price. PSP allows Airnode to act as an oracle and a keeper rolled into one, and provide services that are not available with the oracle + keeper combos that are currently available. From the API integration point-of-view, it imititates an on-chain webhook API, yet does not require the underlying API to be nothing more than a regular REST API.
+
+Even though PSP will enable oracle usage patterns that were not available before, it will also provide a much more permissionless and thus decentralized way of setting up traditional asset price data feeds. In short, one can subscribe to a set of Airnodes to keep individual price feeds up-to-date on a PspBeaconServer contract and update the aggregation when necessary. Although data feeds currently exist, they are not set up through a protocol, but rather through mails being exchanged and meatspace agreements being signed, which is extremely inefficient. Protocolizing setting up first-party oracle-based data feeds across all chains will be a game-changer.
+
+The main difficulty in providing a subscription service is that an API/Airnode can serve a limited number of them at a time (compared to this, RRP is much more scalable). Then, the resource that we must allow the Airnode to manage are these subscription "slots". This slot allocation needs to happen on-chain, programmatically. To achieve this, we implement "allocators", contracts similar to "authorizers" that implement arbitrary business logic that the Airnode operator can choose to opt-in to by adding them in their config file. Note that a requester both needs to have a slot allocated by an allocator, and be authorized by an authorizer to receive service.
+
+The authorizer parameters will be filled as such:
+- `requestId`: `subscriptionId`
+- `airnode`: `airnode`
+- `endpointId`: `endpointId`
+- `sponsor`: `sponsor`
+- `requester`: `fulfillAddress`
+Alternatively, we can refer to the subscriber as the requester (instead of the sponsor), and implement an additional sponsorship layer on top.
```

Comment:
Yeah, I find it strange that the entity calling `subscribe` must be the sponsor, but I'd just rename the contract functions:
- Instead of `subscribe` call it `sponsorSubscription` 
- and `unsponsorSubscription` instead of `unsubscribe`